### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by kasparov92** [facebook/C3D#274]
I am trying the basic example in v1.0, I am having CPU only for now. Following the instructions, I installed cuda so that it works in CPU mode only as I couldn't pass the building phase without the cuda installed! 

When I run the basic example with -1 passed for the GPU, I get the following output: 
`$ sh c3d_sport1m_feature_extraction_frm.sh 
tar (child): v_ApplyEyeMakeup_g01_c01.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
tar (child): v_BaseballPitch_g01_c01.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
NVIDIA: no NVIDIA devices found
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0529 08:44:41.967900 11615 common.cpp:31] Cannot create Cublas handle. Cublas won't be available.
NVIDIA: no NVIDIA devices found
E0529 08:44:41.971379 11615 common.cpp:38] Cannot create Curand generator. Curand won't be available.
E0529 08:44:41.971405 11615 extract_image_features.cpp:68] Using CPU
I0529 08:44:41.991772 11615 net.cpp:76] Creating Layer data
I0529 08:44:41.991796 11615 net.cpp:112] data -> data
I0529 08:44:41.991802 11615 net.cpp:112] data -> label
I0529 08:44:41.991814 11615 video_data_layer.cpp:283] Opening file prototxt/input_list_frm.txt
I0529 08:44:42.014617 11615 video_data_layer.cpp:317] A total of 16 video chunks.
I0529 08:44:42.014648 11615 video_data_layer.cpp:344] read video from input/frm/v_ApplyEyeMakeup_g01_c01/
NVIDIA: no NVIDIA devices found
I0529 08:44:42.300189 11615 video_data_layer.cpp:365] output data size: 50,3,16,112,112
I0529 08:44:42.300225 11615 video_data_layer.cpp:387] Loading mean file from sport1m_train16_128_mean.binaryproto
I0529 08:44:42.370940 11615 net.cpp:127] Top shape: 50 3 16 112 112 (30105600)
I0529 08:44:42.370962 11615 net.cpp:127] Top shape: 50 1 1 1 1 (50)
I0529 08:44:42.370970 11615 net.cpp:159] data does not need backward computation.
I0529 08:44:42.370993 11615 net.cpp:76] Creating Layer conv1a
I0529 08:44:42.370997 11615 net.cpp:86] conv1a <- data
I0529 08:44:42.371007 11615 net.cpp:112] conv1a -> conv1a
I0529 08:44:42.371314 11615 net.cpp:127] Top shape: 50 64 16 112 112 (642252800)
I0529 08:44:42.371320 11615 net.cpp:154] conv1a needs backward computation.
I0529 08:44:42.371335 11615 net.cpp:76] Creating Layer relu1a
I0529 08:44:42.371336 11615 net.cpp:86] relu1a <- conv1a
I0529 08:44:42.371340 11615 net.cpp:100] relu1a -> conv1a (in-place)
I0529 08:44:42.371347 11615 net.cpp:127] Top shape: 50 64 16 112 112 (642252800)
I0529 08:44:42.371351 11615 net.cpp:154] relu1a needs backward computation.
I0529 08:44:42.371354 11615 net.cpp:76] Creating Layer pool1
I0529 08:44:42.371357 11615 net.cpp:86] pool1 <- conv1a
I0529 08:44:42.371361 11615 net.cpp:112] pool1 -> pool1
I0529 08:44:42.371367 11615 net.cpp:127] Top shape: 50 64 16 56 56 (160563200)
I0529 08:44:42.371371 11615 net.cpp:154] pool1 needs backward computation.
I0529 08:44:42.371376 11615 net.cpp:76] Creating Layer conv2a
I0529 08:44:42.371378 11615 net.cpp:86] conv2a <- pool1
I0529 08:44:42.371383 11615 net.cpp:112] conv2a -> conv2a
I0529 08:44:42.373572 11615 net.cpp:127] Top shape: 50 128 16 56 56 (321126400)
I0529 08:44:42.373579 11615 net.cpp:154] conv2a needs backward computation.
I0529 08:44:42.373585 11615 net.cpp:76] Creating Layer relu2a
I0529 08:44:42.373601 11615 net.cpp:86] relu2a <- conv2a
I0529 08:44:42.373607 11615 net.cpp:100] relu2a -> conv2a (in-place)
I0529 08:44:42.373613 11615 net.cpp:127] Top shape: 50 128 16 56 56 (321126400)
I0529 08:44:42.373617 11615 net.cpp:154] relu2a needs backward computation.
I0529 08:44:42.373620 11615 net.cpp:76] Creating Layer pool2
I0529 08:44:42.373622 11615 net.cpp:86] pool2 <- conv2a
I0529 08:44:42.373626 11615 net.cpp:112] pool2 -> pool2
I0529 08:44:42.373631 11615 net.cpp:127] Top shape: 50 128 8 28 28 (40140800)
I0529 08:44:42.373636 11615 net.cpp:154] pool2 needs backward computation.
I0529 08:44:42.373646 11615 net.cpp:76] Creating Layer conv3a
I0529 08:44:42.373648 11615 net.cpp:86] conv3a <- pool2
I0529 08:44:42.373653 11615 net.cpp:112] conv3a -> conv3a
I0529 08:44:42.381562 11615 net.cpp:127] Top shape: 50 256 8 28 28 (80281600)
I0529 08:44:42.381573 11615 net.cpp:154] conv3a needs backward computation.
I0529 08:44:42.381590 11615 net.cpp:76] Creating Layer relu3a
I0529 08:44:42.381594 11615 net.cpp:86] relu3a <- conv3a
I0529 08:44:42.381600 11615 net.cpp:100] relu3a -> conv3a (in-place)
I0529 08:44:42.381605 11615 net.cpp:127] Top shape: 50 256 8 28 28 (80281600)
I0529 08:44:42.381609 11615 net.cpp:154] relu3a needs backward computation.
I0529 08:44:42.381615 11615 net.cpp:76] Creating Layer conv3b
I0529 08:44:42.381618 11615 net.cpp:86] conv3b <- conv3a
I0529 08:44:42.381623 11615 net.cpp:112] conv3b -> conv3b
I0529 08:44:42.397737 11615 net.cpp:127] Top shape: 50 256 8 28 28 (80281600)
I0529 08:44:42.397761 11615 net.cpp:154] conv3b needs backward computation.
I0529 08:44:42.397775 11615 net.cpp:76] Creating Layer relu3b
I0529 08:44:42.397779 11615 net.cpp:86] relu3b <- conv3b
I0529 08:44:42.397785 11615 net.cpp:100] relu3b -> conv3b (in-place)
I0529 08:44:42.397789 11615 net.cpp:127] Top shape: 50 256 8 28 28 (80281600)
I0529 08:44:42.397792 11615 net.cpp:154] relu3b needs backward computation.
I0529 08:44:42.397796 11615 net.cpp:76] Creating Layer pool3
I0529 08:44:42.397799 11615 net.cpp:86] pool3 <- conv3b
I0529 08:44:42.397804 11615 net.cpp:112] pool3 -> pool3
I0529 08:44:42.397811 11615 net.cpp:127] Top shape: 50 256 4 14 14 (10035200)
I0529 08:44:42.397814 11615 net.cpp:154] pool3 needs backward computation.
I0529 08:44:42.397819 11615 net.cpp:76] Creating Layer conv4a
I0529 08:44:42.397822 11615 net.cpp:86] conv4a <- pool3
I0529 08:44:42.397826 11615 net.cpp:112] conv4a -> conv4a
I0529 08:44:42.429550 11615 net.cpp:127] Top shape: 50 512 4 14 14 (20070400)
I0529 08:44:42.429577 11615 net.cpp:154] conv4a needs backward computation.
I0529 08:44:42.429589 11615 net.cpp:76] Creating Layer relu4a
I0529 08:44:42.429602 11615 net.cpp:86] relu4a <- conv4a
I0529 08:44:42.429608 11615 net.cpp:100] relu4a -> conv4a (in-place)
I0529 08:44:42.429613 11615 net.cpp:127] Top shape: 50 512 4 14 14 (20070400)
I0529 08:44:42.429616 11615 net.cpp:154] relu4a needs backward computation.
I0529 08:44:42.429621 11615 net.cpp:76] Creating Layer conv4b
I0529 08:44:42.429625 11615 net.cpp:86] conv4b <- conv4a
I0529 08:44:42.429627 11615 net.cpp:112] conv4b -> conv4b
I0529 08:44:42.492305 11615 net.cpp:127] Top shape: 50 512 4 14 14 (20070400)
I0529 08:44:42.492326 11615 net.cpp:154] conv4b needs backward computation.
I0529 08:44:42.492341 11615 net.cpp:76] Creating Layer relu4b
I0529 08:44:42.492346 11615 net.cpp:86] relu4b <- conv4b
I0529 08:44:42.492359 11615 net.cpp:100] relu4b -> conv4b (in-place)
I0529 08:44:42.492365 11615 net.cpp:127] Top shape: 50 512 4 14 14 (20070400)
I0529 08:44:42.492369 11615 net.cpp:154] relu4b needs backward computation.
I0529 08:44:42.492389 11615 net.cpp:76] Creating Layer pool4
I0529 08:44:42.492393 11615 net.cpp:86] pool4 <- conv4b
I0529 08:44:42.492398 11615 net.cpp:112] pool4 -> pool4
I0529 08:44:42.492404 11615 net.cpp:127] Top shape: 50 512 2 7 7 (2508800)
I0529 08:44:42.492408 11615 net.cpp:154] pool4 needs backward computation.
I0529 08:44:42.492416 11615 net.cpp:76] Creating Layer conv5a
I0529 08:44:42.492420 11615 net.cpp:86] conv5a <- pool4
I0529 08:44:42.492425 11615 net.cpp:112] conv5a -> conv5a
I0529 08:44:42.555388 11615 net.cpp:127] Top shape: 50 512 2 7 7 (2508800)
I0529 08:44:42.555408 11615 net.cpp:154] conv5a needs backward computation.
I0529 08:44:42.555429 11615 net.cpp:76] Creating Layer relu5a
I0529 08:44:42.555433 11615 net.cpp:86] relu5a <- conv5a
I0529 08:44:42.555439 11615 net.cpp:100] relu5a -> conv5a (in-place)
I0529 08:44:42.555444 11615 net.cpp:127] Top shape: 50 512 2 7 7 (2508800)
I0529 08:44:42.555447 11615 net.cpp:154] relu5a needs backward computation.
I0529 08:44:42.555451 11615 net.cpp:76] Creating Layer conv5b
I0529 08:44:42.555455 11615 net.cpp:86] conv5b <- conv5a
I0529 08:44:42.555459 11615 net.cpp:112] conv5b -> conv5b
I0529 08:44:42.617918 11615 net.cpp:127] Top shape: 50 512 2 7 7 (2508800)
I0529 08:44:42.617938 11615 net.cpp:154] conv5b needs backward computation.
I0529 08:44:42.617954 11615 net.cpp:76] Creating Layer relu5b
I0529 08:44:42.617957 11615 net.cpp:86] relu5b <- conv5b
I0529 08:44:42.617962 11615 net.cpp:100] relu5b -> conv5b (in-place)
I0529 08:44:42.617966 11615 net.cpp:127] Top shape: 50 512 2 7 7 (2508800)
I0529 08:44:42.617970 11615 net.cpp:154] relu5b needs backward computation.
I0529 08:44:42.617974 11615 net.cpp:76] Creating Layer pool5
I0529 08:44:42.617977 11615 net.cpp:86] pool5 <- conv5b
I0529 08:44:42.617980 11615 net.cpp:112] pool5 -> pool5
I0529 08:44:42.617985 11615 net.cpp:127] Top shape: 50 512 1 4 4 (409600)
I0529 08:44:42.617987 11615 net.cpp:154] pool5 needs backward computation.
I0529 08:44:42.617991 11615 net.cpp:76] Creating Layer fc6-1
I0529 08:44:42.617995 11615 net.cpp:86] fc6-1 <- pool5
I0529 08:44:42.617997 11615 net.cpp:112] fc6-1 -> fc6-1
I0529 08:44:42.927031 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:42.927053 11615 net.cpp:154] fc6-1 needs backward computation.
I0529 08:44:42.927070 11615 net.cpp:76] Creating Layer relu6
I0529 08:44:42.927075 11615 net.cpp:86] relu6 <- fc6-1
I0529 08:44:42.927080 11615 net.cpp:100] relu6 -> fc6-1 (in-place)
I0529 08:44:42.927084 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:42.927088 11615 net.cpp:154] relu6 needs backward computation.
I0529 08:44:42.927093 11615 net.cpp:76] Creating Layer drop6
I0529 08:44:42.927096 11615 net.cpp:86] drop6 <- fc6-1
I0529 08:44:42.927099 11615 net.cpp:100] drop6 -> fc6-1 (in-place)
I0529 08:44:42.927103 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:42.927105 11615 net.cpp:154] drop6 needs backward computation.
I0529 08:44:42.927109 11615 net.cpp:76] Creating Layer fc7-1
I0529 08:44:42.927112 11615 net.cpp:86] fc7-1 <- fc6-1
I0529 08:44:42.927115 11615 net.cpp:112] fc7-1 -> fc7-1
I0529 08:44:43.092761 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:43.092780 11615 net.cpp:154] fc7-1 needs backward computation.
I0529 08:44:43.092797 11615 net.cpp:76] Creating Layer relu7
I0529 08:44:43.092803 11615 net.cpp:86] relu7 <- fc7-1
I0529 08:44:43.092808 11615 net.cpp:100] relu7 -> fc7-1 (in-place)
I0529 08:44:43.092811 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:43.092814 11615 net.cpp:154] relu7 needs backward computation.
I0529 08:44:43.092819 11615 net.cpp:76] Creating Layer drop7
I0529 08:44:43.092823 11615 net.cpp:86] drop7 <- fc7-1
I0529 08:44:43.092825 11615 net.cpp:100] drop7 -> fc7-1 (in-place)
I0529 08:44:43.092828 11615 net.cpp:127] Top shape: 50 4096 1 1 1 (204800)
I0529 08:44:43.092831 11615 net.cpp:154] drop7 needs backward computation.
I0529 08:44:43.092838 11615 net.cpp:76] Creating Layer fc8-1
I0529 08:44:43.092840 11615 net.cpp:86] fc8-1 <- fc7-1
I0529 08:44:43.092844 11615 net.cpp:112] fc8-1 -> fc8-1
I0529 08:44:43.113353 11615 net.cpp:127] Top shape: 50 487 1 1 1 (24350)
I0529 08:44:43.113373 11615 net.cpp:154] fc8-1 needs backward computation.
I0529 08:44:43.113392 11615 net.cpp:76] Creating Layer prob
I0529 08:44:43.113397 11615 net.cpp:86] prob <- fc8-1
I0529 08:44:43.113404 11615 net.cpp:112] prob -> prob
I0529 08:44:43.113418 11615 net.cpp:127] Top shape: 50 487 1 1 1 (24350)
I0529 08:44:43.113421 11615 net.cpp:154] prob needs backward computation.
I0529 08:44:43.113425 11615 net.cpp:76] Creating Layer accuracy
I0529 08:44:43.113430 11615 net.cpp:86] accuracy <- prob
I0529 08:44:43.113432 11615 net.cpp:86] accuracy <- label
I0529 08:44:43.113436 11615 net.cpp:112] accuracy -> accuracy
I0529 08:44:43.113441 11615 net.cpp:127] Top shape: 1 2 1 1 1 (2)
I0529 08:44:43.113445 11615 net.cpp:154] accuracy needs backward computation.
I0529 08:44:43.113448 11615 net.cpp:165] This network produces output accuracy
I0529 08:44:43.113462 11615 net.cpp:183] Collecting Learning Rate and Weight Decay.
I0529 08:44:43.113483 11615 net.cpp:176] Network initialization done.
I0529 08:44:43.113514 11615 net.cpp:177] Memory required for Data 5653289408
E0529 08:44:47.652730 11615 extract_image_features.cpp:81] Extracting features for 1 batches
Segmentation fault (core dumped)
`

Also, in the runtest, I am getting the following error: 
`build/test/test_all.testbin 0 --gtest_shuffle
NVIDIA: no NVIDIA devices found
Cuda number of devices: 0
Setting to use device 0
Current device id: 0
Note: Randomizing tests' orders with a seed of 7475 .
[==========] Running 417 tests from 74 test cases.
[----------] Global test environment set-up.
[----------] 4 tests from SigmoidCrossEntropyLossLayerTest/1, where TypeParam = double
[ RUN      ] SigmoidCrossEntropyLossLayerTest/1.TestSigmoidCrossEntropyLossGPU
NVIDIA: no NVIDIA devices found
E0529 08:53:32.565585 11941 common.cpp:31] Cannot create Cublas handle. Cublas won't be available.
NVIDIA: no NVIDIA devices found
E0529 08:53:32.600466 11941 common.cpp:38] Cannot create Curand generator. Curand won't be available.
F0529 08:53:32.600548 11941 syncedmem.cpp:53] Check failed: error == cudaSuccess (30 vs. 0)  unknown error
*** Check failure stack trace: ***
    @     0x7f56162255cd  google::LogMessage::Fail()
    @     0x7f5616227433  google::LogMessage::SendToLog()
    @     0x7f561622515b  google::LogMessage::Flush()
    @     0x7f5616227e1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x584350  caffe::SyncedMemory::gpu_data()
    @           0x5eb2c2  caffe::Blob<>::gpu_data()
    @           0x5fd159  caffe::SigmoidLayer<>::Forward_gpu()
    @           0x60012d  caffe::SigmoidCrossEntropyLossLayer<>::Forward_gpu()
    @           0x48cbef  caffe::SigmoidCrossEntropyLossLayerTest<>::TestForward()
    @           0x55c243  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x5556ba  testing::Test::Run()
    @           0x555808  testing::TestInfo::Run()
    @           0x5558e5  testing::TestCase::Run()
    @           0x556bbf  testing::internal::UnitTestImpl::RunAllTests()
    @           0x556ee3  testing::UnitTest::Run()
    @           0x4156cf  main
    @     0x7f560e6f0830  __libc_start_main
    @           0x41a739  _start
    @              (nil)  (unknown)
Makefile:285: recipe for target 'runtest' failed
make: *** [runtest] Aborted (core dumped)
`

I set the CPU only flag to 1, I am not sure what else to be done?

**Comment by kasparov92** [comment]
I decreased the batch size and restarted the machine. It worked
